lib/iris/experimental/ugrid/__init__.py

```diff
@@ -522,6 +522,8 @@ def __eq__(self, other):
             if hasattr(other, "metadata"):
                 # metadata comparison
                 eq = self.metadata == other.metadata
+                if eq:
+                    eq = self.shape == other.shape
```

Comment:
This fixes it so that connectivities of different shapes can compare correctly (i.e. returns False).
Was failing with error, due to problems of array comparison.